packages/diagram/src/module/diagrams/content/uml/sequence/frame.ts

```diff
@@ -0,0 +1,281 @@
+                        // 2. Everything has an individual margin
+                        defaultMarginX = args.marginX
+                        if(defaultMarginX == null) {
+                            defaultMarginX = scope.frameMarginX
+                        }
+                        defaultMarginY = args.marginY
+                        if(defaultMarginY == null) {
+                            defaultMarginY = scope.frameMarginY
+                        }
+                        topLeft = args.topLeft
+                        bottomRight = args.bottomRight
+
+                        marginRight = args.marginRight ?? defaultMarginX
+                        marginBottom = args.marginBottom ?? defaultMarginY
+                        marginLeft = args.marginLeft ?? defaultMarginX
+                        marginTop = args.marginTop ?? defaultMarginY
+
+                        // Calculate the rectangle position
+                        x = topLeft.x - marginLeft
+                        y = topLeft.y - marginTop
+                        width = bottomRight.x - topLeft.x + marginLeft + marginRight
+                        height = bottomRight.y - topLeft.y + marginTop + marginBottom
+
+
+                        // Add the frame
+                        frameElement = canvasElement(
+                            content = rect(class = list("frame")),
+                            class = list("frame-element"),
+                            width = width,
+                            height = height
+                        )
+                        frameElement.topLeft = topLeft
+                        frameElement.bottomRight = bottomRight
+                        frameElement.fragments = list()
+                        frameElement.text = text
+                        frameElement.hasIcon = icon
+                        frameElement.subtext = subtext
+
+                        frameElement.pos = scope.apos(x, y)
+                        frameElement.x = x
+                        frameElement.y = y
+                        frameElement.width = width
+                        frameElement.height = height
+
+                        scope.internal.registerCanvasElement(frameElement, args, args.self)
+
+                        // Lastly, generate all fragments for the given frame, including the implicit top fragment
+                        createSequenceDiagramFragment(topLeft, marginTop = marginTop, parentArgs = args, parent = frameElement, hasIcon = hasIcon, hasLine = false, text = frameText, subtext = subtext)
+                    `,
+                    id("this").assignField(
+                        "sequenceDiagramFragmentFunction",
+                        fun(
+                            `createSequenceDiagramFragment(args[0], marginTop = marginTop, parentArgs = argsCopy, parent = frameElement, hasIcon = args.hasIcon, hasLine = args.hasLine ?? true, text = args.text, subtext = args.subtext)`,
```

Comment:
```suggestion
                            `createSequenceDiagramFragment(it, marginTop = marginTop, parentArgs = argsCopy, parent = frameElement, hasIcon = args.hasIcon, hasLine = args.hasLine ?? true, text = args.text, subtext = args.subtext)`,
```

Reply:
Why? Is `args[0]` always assigned to be `it`?

Reply:
the other way around
it is the alias for args[0] (and used basically everywhere else instead of args[0])